Removal of deprecated astropy.tests.pytest_plugins from conftest.py L9

Hi,

My recent commits in PR #659 passed almost all the tests that should be have been passed except for the Travis CI. Upon closer inspection of the code, I found out that there is some problem with the following file: 

[https://github.com/astropy/ccdproc/blob/master/ccdproc/conftest.py](https://github.com/astropy/ccdproc/blob/master/ccdproc/conftest.py)

The error thrown in the output of the Travis CI is as follows:

```
ImportError while loading conftest '/tmp/ccdproc-test-r9jki1ep/lib/python3.6/site-packages/ccdproc/conftest.py'.
ccdproc/conftest.py:9: in <module>
    from astropy.tests.pytest_plugins import *
E   ModuleNotFoundError: No module named 'astropy.tests.pytest_plugins'
The command "$MAIN_CMD $SETUP_CMD" exited with 4.
```

([https://travis-ci.org/astropy/ccdproc/jobs/493581671](https://travis-ci.org/astropy/ccdproc/jobs/493581671))

Is there anything I can do about the situation? 

I think this Issue is related to astropy/astropy#7782...

--- a/.travis.yml
+++ b/.travis.yml
@@ -24,8 +24,8 @@ env:
         # to repeat them for all configurations.
         - NUMPY_VERSION=stable
         - ASTROPY_VERSION=stable
-        - CONDA_DEPENDENCIES='scipy astroscrappy reproject scikit-image psutil'
-        - PIP_DEPENDENCIES=''
+        - CONDA_DEPENDENCIES='scipy reproject scikit-image psutil'
+        - PIP_DEPENDENCIES='git+https://github.com/astropy/astroscrappy.git'
         - MAIN_CMD='python setup.py'
         - SETUP_CMD='test'
         - CONDA_CHANNELS='astropy-ci-extras astropy'

--- a/.travis.yml
+++ b/.travis.yml
@@ -24,8 +24,11 @@ env:
         # to repeat them for all configurations.
         - NUMPY_VERSION=stable
         - ASTROPY_VERSION=stable
-        - CONDA_DEPENDENCIES='scipy reproject scikit-image psutil cython'
-        - PIP_DEPENDENCIES='git+https://github.com/astropy/astroscrappy.git'
+        - CONDA_DEPENDENCIES='scipy reproject psutil cython astroscrappy'
+
+        # Need the latest scikit-image (0.14.2 or higher), which is not in
+        # anaconda yet.
+        - PIP_DEPENDENCIES='scikit-image'
         - MAIN_CMD='python setup.py'
         - SETUP_CMD='test'
         - CONDA_CHANNELS='astropy-ci-extras astropy'

--- a/.travis.yml
+++ b/.travis.yml
@@ -24,7 +24,7 @@ env:
         # to repeat them for all configurations.
         - NUMPY_VERSION=stable
         - ASTROPY_VERSION=stable
-        - CONDA_DEPENDENCIES='scipy reproject scikit-image psutil'
+        - CONDA_DEPENDENCIES='scipy reproject scikit-image psutil cython'
         - PIP_DEPENDENCIES='git+https://github.com/astropy/astroscrappy.git'
         - MAIN_CMD='python setup.py'
         - SETUP_CMD='test'

--- a/.travis.yml
+++ b/.travis.yml
@@ -16,6 +16,7 @@ python:
     - 3.4
     - 3.5
     - 3.6
+    - 3.7
 
 env:
     global:
@@ -40,17 +41,17 @@ matrix:
 
         # Try Astropy/NumPy development versions. This requires them to be
         # compiled during setup which takes some time.
-        - python: 3.6
+        - python: 3.7
           env: ASTROPY_VERSION=development
                NUMPY_VERSION=dev
                SETUP_CMD='test --coverage'
 
         # Check for sphinx doc build warnings
-        - python: 3.6
+        - python: 3.7
           env: SETUP_CMD='build_docs -w'
 
         # Do coverage tests for both latest Python versions
-        - python: 3.6
+        - python: 3.7
           env: ASTROPY_VERSION=3
                SETUP_CMD='test --coverage'
 
@@ -71,11 +72,11 @@ matrix:
 
         # Try numpy pre-release version. This runs only when a pre-release
         # is available on pypi.
-        - python: 3.6
+        - python: 3.7
           env: NUMPY_VERSION=prerelease
 
         # pycodestyle test, see setup.cfg for tested pep8 warnings/errors
-        - python: 3.6
+        - python: 3.7
           env: MAIN_CMD='pycodestyle ccdproc --count  --show-source --show-pep8'
                SETUP_CMD=''
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -16,7 +16,6 @@ python:
     - 3.4
     - 3.5
     - 3.6
-    - 3.7
 
 env:
     global:
@@ -33,26 +32,27 @@ env:
         - ASTROPY_USE_SYSTEM_PYTEST=1
 
     matrix:
-
         - SETUP_CMD='egg_info'
 
 matrix:
     include:
+        - env: PYTHON_VERSION=3.7
+               SETUP_CMD='egg_info'
 
         # Try Astropy/NumPy development versions. This requires them to be
         # compiled during setup which takes some time.
-        - python: 3.7
-          env: ASTROPY_VERSION=development
+        - env: PYTHON_VERSION=3.7
+               ASTROPY_VERSION=development
                NUMPY_VERSION=dev
                SETUP_CMD='test --coverage'
 
         # Check for sphinx doc build warnings
-        - python: 3.7
-          env: SETUP_CMD='build_docs -w'
+        - env: PYTHON_VERSION=3.7
+               SETUP_CMD='build_docs -w'
 
         # Do coverage tests for both latest Python versions
-        - python: 3.7
-          env: ASTROPY_VERSION=3
+        - env: PYTHON_VERSION=3.7
+               ASTROPY_VERSION=3
                SETUP_CMD='test --coverage'
 
         # Check compatibility with the current astropy LTS release
@@ -72,12 +72,12 @@ matrix:
 
         # Try numpy pre-release version. This runs only when a pre-release
         # is available on pypi.
-        - python: 3.7
-          env: NUMPY_VERSION=prerelease
+        - env: PYTHON_VERSION=3.7
+               NUMPY_VERSION=prerelease
 
         # pycodestyle test, see setup.cfg for tested pep8 warnings/errors
-        - python: 3.7
-          env: MAIN_CMD='pycodestyle ccdproc --count  --show-source --show-pep8'
+        - env: PYTHON_VERSION=3.7
+               MAIN_CMD='pycodestyle ccdproc --count  --show-source --show-pep8'
                SETUP_CMD=''
 
 install:

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -10,8 +10,8 @@ environment:
       PYTHON_ARCH: "64" # needs to be set for CMD_IN_ENV to succeed. If a mix
                         # of 32 bit and 64 bit builds are needed, move this
                         # to the matrix section.
-      CONDA_DEPENDENCIES: "scipy astroscrappy reproject scikit-image"
-      PIP_DEPENDENCIES: ""
+      CONDA_DEPENDENCIES: "scipy reproject scikit-image"
+      PIP_DEPENDENCIES: "git+https://github.com/astropy/astroscrappy.git"
       CONDA_CHANNELS: "astropy"
       ASTROPY_USE_SYSTEM_PYTEST: 1
 

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -16,7 +16,7 @@ environment:
       ASTROPY_USE_SYSTEM_PYTEST: 1
 
   matrix:
-      - PYTHON_VERSION: "3.6"
+      - PYTHON_VERSION: "3.7"
         ASTROPY_VERSION: "stable"
         NUMPY_VERSION: "stable"
 

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -10,7 +10,7 @@ environment:
       PYTHON_ARCH: "64" # needs to be set for CMD_IN_ENV to succeed. If a mix
                         # of 32 bit and 64 bit builds are needed, move this
                         # to the matrix section.
-      CONDA_DEPENDENCIES: "scipy reproject scikit-image"
+      CONDA_DEPENDENCIES: "scipy reproject scikit-image cython"
       PIP_DEPENDENCIES: "git+https://github.com/astropy/astroscrappy.git"
       CONDA_CHANNELS: "astropy"
       ASTROPY_USE_SYSTEM_PYTEST: 1

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -10,8 +10,10 @@ environment:
       PYTHON_ARCH: "64" # needs to be set for CMD_IN_ENV to succeed. If a mix
                         # of 32 bit and 64 bit builds are needed, move this
                         # to the matrix section.
-      CONDA_DEPENDENCIES: "scipy reproject scikit-image cython"
-      PIP_DEPENDENCIES: "git+https://github.com/astropy/astroscrappy.git"
+      CONDA_DEPENDENCIES: "scipy reproject cython astroscrappy"
+      # Need the latest scikit-image (0.14.2 or higher), which is not in
+      # anaconda yet.
+      PIP_DEPENDENCIES: "scikit-image"
       CONDA_CHANNELS: "astropy"
       ASTROPY_USE_SYSTEM_PYTEST: 1
 

--- a/ccdproc/conftest.py
+++ b/ccdproc/conftest.py
@@ -16,6 +16,14 @@ except ImportError:
                                               PYTEST_HEADER_MODULES,
                                               TESTED_VERSIONS)
 
+try:
+    # This is the way to get plugins in astropy 2.x
+    from astropy.tests.pytest_plugins import *
+except ImportError:
+    # Otherwise they are installed as separate packages that pytest
+    # automagically finds.
+    pass
+
 from .tests.pytest_fixtures import *
 
 # This is to figure out ccdproc version, rather than using Astropy's

--- a/ccdproc/conftest.py
+++ b/ccdproc/conftest.py
@@ -6,7 +6,16 @@
 
 import os
 
-from astropy.tests.pytest_plugins import *
+try:
+    from astropy.tests.plugins.display import (pytest_report_header,
+                                               PYTEST_HEADER_MODULES,
+                                               TESTED_VERSIONS)
+except ImportError:
+    # When using astropy 2.0
+    from astropy.tests.pytest_plugins import (pytest_report_header,
+                                              PYTEST_HEADER_MODULES,
+                                              TESTED_VERSIONS)
+
 from .tests.pytest_fixtures import *
 
 # This is to figure out ccdproc version, rather than using Astropy's

